README.md

```diff
@@ -279,10 +306,12 @@ The result can be used like this:
 }
 ```
 
-Put reusable nix functions that are intend for public use in the `lib` attribute:
+### Providing library functions
+
+Put reusable nix functions that are intend for public use in the `lib` attribute. Make sure to use the given `lib` argument instead of `pkgs.lib` if you depend on nixpkgs functions.
 
 ```nix
-{ lib }:
+{ pkgs, lib }:
```

Comment:
Is it necessary to pass `lib` if it's accessible through `pkgs.lib`?

Reply:
Using `lib` without a dependency on `pkgs` means that it's possible to use library functions (which are built only via `lib`) in your NixOS config without having to worry about infinite recursion, e.g. when using such a function in an overlay (using `pkgs.lib` would give inf rec because pkgs depends on overlays).

Reply:
In an overlay `lib` from `super` is supposed to be [used](https://blog.flyingcircus.io/2017/11/07/nixos-the-dos-and-donts-of-nixpkgs-overlays/) and for nixos modules, `lib` from the function argument is preferred. Would the current approach bypass overlays by reimporting `nixpkgs`? Are there any other use cases that I have not covered yet?